bin/reth/src/commands/import_receipts.rs

```diff
@@ -97,47 +92,83 @@ impl ImportReceiptsCommand {
             );
         }
 
+        // prepare the tx for `write_to_storage`
         let tx = provider.into_tx();
+        let mut total_decoded_receipts = 0;
+
+        // open file
+        let mut reader = ChunkedFileReader::new(&self.path, self.chunk_len).await?;
 
         while let Some(file_client) = reader.next_chunk::<ReceiptFileClient>().await? {
             // create a new file client from chunk read from file
+            let ReceiptFileClient {
+                mut receipts,
+                mut first_block,
+                total_receipts: total_receipts_chunk,
+            } = file_client;
+
+            // mark these as decoded
+            total_decoded_receipts += total_receipts_chunk;
+
             info!(target: "reth::cli",
+                first_receipts_block=?first_block,
+                total_receipts_chunk,
                 "Importing receipt file chunk"
             );
 
-            let ReceiptFileClient { receipts, first_block, total_receipts: total_receipts_chunk } =
-                file_client;
-
-            total_decoded_receipts += total_receipts_chunk;
-
+            // It is possible for the first receipt returned by the file client to be block #1
+            // rather than the genesis block. In this case, we just prepend empty receipts to the
+            // current list of receipts. When initially writing to static files, the provider
+            // expects the first block to be block zero. So, if the first block returned by the file
+            // client is block #1, we insert empty receipts for the genesis block.
+            if first_block == 1 {
+                // prepend the first empty receipts
+                receipts.insert(0, vec![]);
+                // this ensures the bundle state and static file producer start at zero
+                first_block = 0;
+                // we count this as decoded so the partial import check later does not error if this
+                // branch is executed
+                total_decoded_receipts += 1;
+            }
+
+            // We're reusing receipt writing code internal to
+            // `BundleStateWithReceipts::write_to_storage`, so we just use a default empty
+            // `BundleState`.
             let bundled_state =
                 BundleStateWithReceipts::new(Default::default(), receipts, first_block);
 
             let static_file_producer =
                 static_file_provider.get_writer(first_block, StaticFileSegment::Receipts)?;
 
+            // finally, write the receipts
             bundled_state.write_to_storage::<<DatabaseEnv as Database>::TXMut>(
                 &tx,
                 Some(static_file_producer),
                 OriginalValuesKnown::Yes,
             )?;
 
-            static_file_provider.commit()?
+            static_file_provider.commit()?;
```

Comment:
lets put it outside of the loop instead

as static files works in file ranges, internally it will be committing when creating the next file range already, so we only need to call explicitly at the end.